R/FletcherOEData.R

```diff
@@ -0,0 +1,69 @@
+#' Obtain the Fletcher OE data
+#'
+#' Obtain the mouse olfactory epithelial HBC stem cell differentiation dataset
+#' from Fletcher et al. (2017).
+#'
+#' @param filtered Logical scalar indicating whether to filter out cells that
+#'   were not used by the authors.
+#' @param ensembl Logical scalar indicating whether the output row names should
+#'   contain Ensembl identifiers.
+#' @param location Logical scalar indicating whether genomic coordinates should
+#'   be returned.
+#' @details
+#' Column metadata is scraped from GEO, using both the author-supplied
+#' "phenoData" per-cell annotations and the author-supplied "protocolData"
+#' per-cell annotations. The former includes information about the animals and
+#' the instruments, while the latter contains QC statistics.
+#' 
+#' We also included the clustering results from the authors' analysis.
+#'
+#' If \code{filtered=TRUE}, only the cells used by the authors in their cluster
+#' analysis are returned. Otherwise, the cells not used by the authors will have
+#' NA in the clustering columns of the \code{\link{colData}}.
+#'
+#' If \code{ensembl=TRUE}, the gene symbols are converted to Ensembl IDs in the
+#' row names of the output object. Rows with missing Ensembl IDs are discarded,
+#' and only the first occurrence of duplicated IDs is retained.
+#'
+#' If \code{location=TRUE}, the coordinates of the Ensembl gene models are
+#' stored in the \code{\link{rowRanges}} of the output.
+#' 
+#' All data are downloaded from ExperimentHub and cached for local re-use.
+#' Specific resources can be retrieved by searching for
+#' \code{scRNAseq/fletcher-oe}.
+#'
+#' @return A \linkS4class{SingleCellExperiment} object with a single matrix of
+#'   read counts.
+#'
+#' @author Davide Risso
+#'
+#' @references
+#' Fletcher et al. (2017). Deconstructing Olfactory Stem
+#' Cell Trajectories at Single-Cell Resolution. \emph{Cell Stem Cell} 20(6):
+#' 817–30.
+#'
+#' @examples
+#' sce <- FletcherOEData()
+#' 
+#' @export
+#' @importFrom SingleCellExperiment splitAltExps altExp altExp<-
+#' @importFrom SummarizedExperiment rowData rowData<-
+FletcherOEData <- function(filtered=TRUE, ensembl=FALSE, location=TRUE) {
+    version <- "2.6.0"
+
+    sce <- .create_sce(file.path("fletcher-oe", version), has.rowdata=TRUE)
+
+    if (filtered) {
+        sce <- sce[,!is.na(sce$cluster_id)]
```

Comment:
I would prefer we had a `retained` TRUE/FALSE column for consistency with others, e.g., `JessaBrainData`. In principle, a `NA` field in the clustering could mean other things, e.g., the cell was retained but the authors didn't know what it was.

Reply:
I'm not sure that I understand, doens't `JessaBrainData` also have a `filtered=TRUE` argument?

Reply:
Yes, and it also has an explicit `retained` column indicating whether that cell was retained.

Reply:
Oh I see what you mean now, I'll add the column